mixer/adapter/rbac/rbacStore.go

```diff
@@ -0,0 +1,233 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"strings"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/authorization"
+)
+
+// Rbac interface
+type Rbac interface {
+	CheckPermission(inst *authorization.Instance, env adapter.Env) (bool, error)
+}
+
+// Information about a ServiceRole and associted ServiceRoleBindings
+type roleInfo struct {
+	// ServiceRole proto definition
+	info *rbacproto.ServiceRole
+
+	// A set of ServiceRoleBindings that refer to this role.
+	bindings map[string]*rbacproto.ServiceRoleBinding
+}
+
+// maps role name to role info
+type rolesByName map[string]*roleInfo
+
+// maps namespace to a set of roles in the namespace
+type rolesMapByNamespace map[string]rolesByName
+
+// configStore contains all ServiceRole and ServiceRoleBinding information.
+// configStore implements Rbac interface.
+type configStore struct {
+	// All the roles organized per namespace.
+	roles rolesMapByNamespace
+}
+
+// Create a RoleInfo object.
+func newRoleInfo(spec *rbacproto.ServiceRole) *roleInfo {
+	return &roleInfo{
+		info: spec,
+	}
+}
+
+// Set a binding for a given Service role.
+func (ri *roleInfo) setBinding(name string, spec *rbacproto.ServiceRoleBinding) {
+	if ri == nil {
+		return
+	}
+	if ri.bindings == nil {
+		ri.bindings = make(map[string]*rbacproto.ServiceRoleBinding)
+	}
+	ri.bindings[name] = spec
+}
+
+// Update roles in the RBAC store.
+func (rs *configStore) changeRoles(roles rolesMapByNamespace) {
+	rs.roles = roles
+}
+
+// CheckPermission checks permission for a given request. This is the main API called
+// by RBAC adapter at runtime to authorize requests.
+func (rs *configStore) CheckPermission(inst *authorization.Instance, env adapter.Env) (bool, error) {
+	namespace := inst.Action.Namespace
+	if namespace == "" {
+		env.Logger().Errorf("Missing namespace")
+		return false, nil
+	}
+
+	serviceName := inst.Action.Service
+	if serviceName == "" {
+		env.Logger().Errorf("Missing service")
+		return false, nil
+	}
+
+	path := inst.Action.Path
+	if path == "" {
+		env.Logger().Errorf("Missing path")
+		return false, nil
+	}
+
+	method := inst.Action.Method
+	if method == "" {
+		env.Logger().Errorf("Missing method")
+		return false, nil
+	}
+
+	extraAct := inst.Action.Properties
+
+	user := inst.Subject.User
+	groups := inst.Subject.Groups
+
+	instSub := inst.Subject.Properties
+
+	rn := rs.roles[namespace]
+	if rn == nil {
+		return false, nil
+	}
+
+	for rolename, roleInfo := range rn {
+		eligibleRole := false
+		env.Logger().Infof("Checking role: %s", rolename)
```

Comment:
@geeknoid this feels like debug level info. what is the current logging recommendation for adapter env level debug logging?